scripts/core/editor3/components/spellchecker/default-spellcheckers.tsx

```diff
@@ -33,29 +33,89 @@ function check(str: string): Promise<Array<ISpellcheckWarning>> {
-                let matchArr;
-                let start;
-
-                // tslint:disable-next-line no-conditional-assignment
-                while ((matchArr = regex.exec(paragraph)) !== null) {
-                    start = matchArr.index;
-                    if (!spellcheck.isCorrectWord(matchArr[0])) {
-                        info.push({
-                            startOffset: lastOffset + start,
-                            text: matchArr[0],
-                            suggestions: null,
-                        });
-                    }
-                }
+            let matchArr;
+            let start;
 
-                lastOffset += paragraph.length;
-            });
+            // tslint:disable-next-line no-conditional-assignment
+            while ((matchArr = regex.exec(str)) !== null) {
+                start = matchArr.index;
+                if (!spellcheck.isCorrectWord(matchArr[0])) {
+                    info.push({
+                        startOffset: lastOffset + start,
+                        text: matchArr[0],
+                        suggestions: null,
+                        type: 'spelling',
+                    });
+                }
+            }
 
             return info;
         });
 }
 
 export function getSpellchecker(language: string): ISpellchecker {
-    return {check, getSuggestions, actions};
+    const spellcheckerName = ({
+        fr: 'grammalecte',
+        nl: 'leuven_dutch',
+    })[language];
+
+    if (spellcheckerName != null) {
+        return {
+            check: (str: string) => ng.getServices(['config', 'session']).then((services: Array<any>) => {
+                const [config, session] = services;
+
+                return new Promise((resolve, reject) => {
+                    const xhr = new XMLHttpRequest();
+
+                    xhr.open('POST', config.server.url + '/spellchecker', true);
+
+                    xhr.setRequestHeader('Content-Type', 'application/json');
+                    xhr.setRequestHeader('Authorization', session.token);
+
+                    xhr.onload = function() {
+                        if (this.status >= 200 && this.status <= 300) {
+                            try {
+                                resolve(JSON.parse(this.responseText).errors);
+                            } catch (e) {
+                                reject('invalid JSON received from server');
+                            }
+                        } else {
+                            reject('server error');
+                        }
+                    };
+
+                    xhr.send(JSON.stringify({spellchecker: spellcheckerName, text: str}));
+                });
+            }),
+            getSuggestions: (str) => ng.getServices(['config', 'session']).then((services: Array<any>) => {
+                const [config, session] = services;
+
+                return new Promise((resolve, reject) => {
+                    const xhr = new XMLHttpRequest();
```

Comment:
this xhr handling is duplicated in both methods, and imo it shouldn't be in spellchecker anyhow